docs/_blog/2020-04-16-working-with-private-registries.md

```diff
@@ -0,0 +1,139 @@
+
+*Hello there, general Kenobi, this is codewind saying hi and thanks for using me*
+{:style="margin: 30px; text-align: center;"}
+
+Much better.
+
+At this point, I'd like to highlight the structure of the `Dockerfile` with respect to what is actually happening with the `.npmrc` file. The relevant lines in the `Dockerfile` are:
+
+```Dockerfile
+# Install npm production packages
+COPY package.json .npmrc ./
+RUN npm install --production
+
+# Copy application source
+COPY . ./
+RUN rm .npmrc
+```
+{:style="margin: 30px;"}
+
+This is the last piece of the puzzle--as you can see, the `.npmrc` file is copied to the Docker build for the `npm install` and removed afterwards. The build is also a multi-stage one, which ensures the `.npmrc` file is not leaked in an intermediate layer of the final Docker image. **Note:** if you are trying this exercise yourself, make sure you have the latest multi-stage version of the `Dockerfile` [here](https://github.com/codewind-resources/nodeExpressTemplate/blob/master/Dockerfile).
+
+### A Java Maven Example
+
+The steps to configure and pull from a private Maven repository for Java projects is very similar to the NPM example I've described, so I won't go into as much detail here. Once again, I start by setting up the proper credentials to access the private server; for Maven, that means adding a `server` element to my `~/.m2/settings.xml` file:
+
+```xml
+<server>
+  <id>github</id>
+  <username>makandre</username>
+  <password>****************************************</password>
+</server>
+```
+{:style="margin: 30px;"}
+
+And now to add the repository location and dependency I need to my project's `pom.xml` file:
+
+```xml
+<repositories>
+  <repository>
+    <id>github</id>
+    <url>https://maven.pkg.github.com/makandre/privateRegistry</url>
+  </repository>
+</repositories>
+
+<pluginRepositories>
+  <pluginRepository>
+    <id>github</id>
+    <url>https://maven.pkg.github.com/makandre/privateRegistry</url>
+  </pluginRepository>
+<pluginRepositories>
+
+<dependencies>
+  <dependency>
+    <groupId>org.eclipse.codewind</groupId>
+    <artifactId>sampleJar</artifactId>
+    <version>0.0.1</version>
+  </dependency>
+</dependencies>
+```
+{:style="margin: 30px;"}
+
+The last step is to add a [reference](referencing-files.html) to my `~/.m2/settings.xml` file. Now my project should be all set to pull the `sampleJar` dependency down from the configured private repository.
```

Comment:
Could we remove the word "down"? It will likely get flagged in our accessibility checker as a directional word, which might be confusing for users with screen readers. I know in this context it's not confusing...but removing it will give us one less thing to get picked up by the accessibility checker. :)

Reply:
Changed to "download"